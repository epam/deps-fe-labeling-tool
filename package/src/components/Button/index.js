import { Button, ButtonType } from './Button'
import { IconButton } from './IconButton'
import { TextButton } from './TextButton.styles'

Button.Icon = IconButton

export {
  Button,
  ButtonType,
  TextButton
}
