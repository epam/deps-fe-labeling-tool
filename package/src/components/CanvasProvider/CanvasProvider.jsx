import React, { PureComponent } from 'react'
import { canvasShape } from '@/models/Canvas'

const CANVAS_HOC_NAME = 'CanvasHOC'

const withCanvas = (WrapperComponent) => {
  return class CanvasHOC extends PureComponent {
    static propTypes = {
      canvas: canvasShape
    }

    static displayName = CANVAS_HOC_NAME

    render = () => {
      if (!this.props.canvas) {
        return null
      }

      return <WrapperComponent {...this.props} />
    }
  }
}

export {
  CANVAS_HOC_NAME,
  withCanvas
}
