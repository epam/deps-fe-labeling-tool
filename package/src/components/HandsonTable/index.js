
export * from './models'
export * from './HandsonTable'
