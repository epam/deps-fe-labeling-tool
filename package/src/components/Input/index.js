import { FilterInput } from './FilterInput'
import { Input } from './Input'
import { NumericInput } from './NumericInput'

Input.Numeric = NumericInput
Input.Filter = FilterInput

export {
  Input
}
