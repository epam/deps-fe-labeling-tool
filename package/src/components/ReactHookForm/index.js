export * from './Form'
export * from './FormField'
export * from './FormItem'
export * from './Validators'
export * from './FormFieldType'
