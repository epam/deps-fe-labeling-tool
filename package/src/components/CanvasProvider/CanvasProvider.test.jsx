import React from 'react'
import { shallow } from 'enzyme'
import { withCanvas } from '@/components/CanvasProvider'

const WithCanvas = withCanvas(() => null)

const MOCK_CANVAS = {}

describe('HOC: CanvasProvider', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<WithCanvas canvas={MOCK_CANVAS} />)
  })

  it('should render wrapped component if canvas was provided', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render anything if canvas was not provided', () => {
    wrapper.setProps({
      canvas: undefined
    })

    expect(wrapper.children().exists()).toBeFalsy()
  })
})
