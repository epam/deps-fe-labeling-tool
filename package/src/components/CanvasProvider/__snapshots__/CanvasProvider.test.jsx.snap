// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC: CanvasProvider should render wrapped component if canvas was provided 1`] = `
<Component
  canvas={Object {}}
/>
`;
